it/src/main/java/com/google/cloud/teleport/it/pubsub/DefaultPubsubResourceManager.java

```diff
@@ -0,0 +1,253 @@
+    checkIsUsable();
+
+    LOG.info("Creating topic '{}'...", topicName);
+
+    Topic topic = topicAdminClient.createTopic(getTopicName(topicName));
+    TopicName reference = PubsubUtils.toTopicName(topic);
+    createdTopics.add(reference);
+
+    LOG.info("Topic '{}' was created successfully!", topicName);
+
+    return reference;
+  }
+
+  /**
+   * Create a subscription based on the given name. However, it adds a prefix based on the test id
+   * and timestamp, to uniquely identify the current execution instance.
+   *
+   * @see PubsubResourceManager#createSubscription(TopicName, String)
+   */
+  @Override
+  public SubscriptionName createSubscription(TopicName topicName, String subscriptionName) {
+    checkArgument(!subscriptionName.isEmpty(), "subscriptionName can not be empty");
+    checkIsUsable();
+
+    if (!createdTopics.contains(topicName)) {
+      throw new IllegalArgumentException(
+          "Can not create a subscription for a topic not managed by this instance.");
+    }
+
+    LOG.info("Creating subscription '{}' for topic '{}'", subscriptionName, topicName);
+
+    Subscription subscription =
+        subscriptionAdminClient.createSubscription(
+            getSubscriptionName(subscriptionName),
+            topicName,
+            PushConfig.getDefaultInstance(),
+            DEFAULT_ACK_DEADLINE_SECONDS);
+    SubscriptionName reference = PubsubUtils.toSubscriptionName(subscription);
+    createdSubscriptions.add(reference);
+
+    LOG.info(
+        "Subscription '{}' for topic '{}' was created successfully!", topicName, subscriptionName);
+
+    return reference;
+  }
+
+  @Override
+  public String publish(TopicName topic, Map<String, String> attributes, ByteString data)
+      throws PubsubResourceManagerException {
+    checkIsUsable();
+
+    if (!createdTopics.contains(topic)) {
+      throw new IllegalArgumentException(
+          "Can not publish to a topic not managed by this instance.");
+    }
+
+    LOG.info("Publishing message with {} bytes to topic '{}'", data.size(), topic);
+
+    PubsubMessage pubsubMessage =
+        PubsubMessage.newBuilder().putAllAttributes(attributes).setData(data).build();
+
+    try {
+      String messageId = publisherFactory.createPublisher(topic).publish(pubsubMessage).get();
+      LOG.info("Message published with id '{}'", messageId);
+      return messageId;
+    } catch (Exception e) {
+      throw new PubsubResourceManagerException("Error publishing message to Pubsub", e);
+    }
+  }
+
+  @Override
+  public synchronized void cleanupAll() {
+    // Ignore call if it was cleaned up before
+    if (isNotUsable()) {
+      return;
+    }
+
+    LOG.info("Attempting to cleanup manager.");
+
+    try {
+      for (SubscriptionName subscription : createdSubscriptions) {
+        LOG.info("Deleting subscription '{}'", subscription);
+        subscriptionAdminClient.deleteSubscription(subscription);
+      }
+    } finally {
+      subscriptionAdminClient.close();
```

Comment:
I think the two try blocks should be combined so that we can make sure both clients are closed in the same finally block.